### PR TITLE
Adjust for SE-0065 "A New Model for Collections and Indices"

### PR DESCRIPTION
**Pull request by gribozavr**
This gets swiftpm building, but we can't run tests because XCTest is not ported yet.

### CODE REVIEW

File: Sources/ManifestParser/TOML.swift [hunk 1]
```diff
@@ -83,27 +83,19 @@ public func ==(lhs: TOMLItem, rhs: TOMLItem) -> Bool {
...
 
     /// Check if this is a valid initial character of a number constant.
     func isNumberInitialChar() -> Bool {
         switch self {
-        case "+".utf8Constant: fallthrough
-        case "-".utf8Constant: fallthrough
-        case "0".utf8Constant..."9".utf8Constant:
+        case UInt8(ascii: "+"),
+             UInt8(ascii: "-"),
+             UInt8(ascii: "0")...UInt8(ascii:"9"):
```
Comment (ddunbar) [inline Sources/ManifestParser/TOML.swift]: Thanks! :)

File: Sources/Utility/Path.swift [hunk 2]
```diff
@@ -290,12 +290,12 @@ extension String {
...
             while idx != end {
                 if utf8[idx] == UInt8(ascii: "/") {
-                    idx = idx.successor()
+                    utf8.formLocation(after: &idx)
                     if idx == end || utf8[idx] == UInt8(ascii: "/") {
                         return false
                     }
                 }
-                idx = idx.successor()
+                utf8.formLocation(after: &idx)
```
Comment (ddunbar) [inline Sources/Utility/Path.swift]: What is the point of `formLocation` versus just `foo = location(after:)`?

**Comment by ddunbar** [comment]
LGTM, thanks!

File: Sources/Utility/Path.swift (hunk 2 above)
Comment (gribozavr) [inline Sources/Utility/Path.swift]: `formLocation()` can perform the operation in place, and thus can be cheaper when indices are non-trivial.  In this case though, the index for a string view will eventually become a wrapped `Int`, so there would be no difference.  I can revert to the assignment if you prefer.

Comment (ddunbar) [inline Sources/Utility/Path.swift]: Nope, was just curious about the API.

**Comment by gribozavr** [comment]
All tests pass now.

**Pull request merged**